includes/REST_API/AI_Prompt_REST_Controller.php

```diff
@@ -0,0 +1,321 @@
+					'required'    => true,
+					'minItems'    => 1,
+				),
+				'modelConfig'      => JSON_Schema_To_WP_Schema_Converter::convert( ModelConfig::getJsonSchema() ),
+				'providerId'       => array(
+					'description' => __( 'The provider ID, to enforce using a model from that provider.', 'wp-ai-client' ),
+					'type'        => 'string',
+				),
+				'modelId'          => array(
+					'description' => __( 'The model ID, to enforce using that model. If given, a providerId must also be present.', 'wp-ai-client' ),
+					'type'        => 'string',
+				),
+				'modelPreferences' => array(
+					'description' => __( 'List of preferred models.', 'wp-ai-client' ),
+					'type'        => 'array',
+					'items'       => array(
+						'oneOf' => array(
+							array(
+								'type' => 'string',
+							),
+							array(
+								'type'     => 'array',
+								'items'    => array(
+									'type' => 'string',
+								),
+								'minItems' => 2,
+								'maxItems' => 2,
+							),
+						),
+					),
+				),
+				'capability'       => array(
+					'description' => __( 'The capability to use.', 'wp-ai-client' ),
+					'type'        => 'string',
+					'enum'        => CapabilityEnum::getValues(),
+				),
+				'requestOptions'   => JSON_Schema_To_WP_Schema_Converter::convert( RequestOptions::getJsonSchema() ),
+			),
+		);
+	}
+
+	/**
+	 * Retrieves the generation result schema.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array<string, mixed> The result schema.
+	 */
+	public function get_generation_result_schema(): array {
+		$schema            = GenerativeAiResult::getJsonSchema();
+		$schema['$schema'] = 'http://json-schema.org/draft-04/schema#';
+		$schema['title']   = 'ai_generation_result';
+
+		return JSON_Schema_To_WP_Schema_Converter::convert( $schema );
+	}
+
+	/**
+	 * Retrieves the supported check schema.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array<string, mixed> The supported check schema.
+	 */
+	public function get_is_supported_schema(): array {
+		return array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'ai_is_supported_response',
+			'type'       => 'object',
+			'properties' => array(
+				'supported' => array(
+					'description' => __( 'Whether the capability is supported.', 'wp-ai-client' ),
+					'type'        => 'boolean',
+					'required'    => true,
+				),
+			),
+		);
+	}
+
+	/**
+	 * Helper to create builder from params.
+	 *
+	 * @param array<string, mixed> $params The parameters.
+	 * @phpstan-param GenerationRequestParams $params
+	 * @return Prompt_Builder The builder instance.
+	 */
+	private function create_builder_from_params( array $params ): Prompt_Builder {
+		// Messages are required by schema.
+		$messages_data = $params['messages'];
+
+		$messages = array_map(
+			fn( $message ) => Message::fromArray( $message ),
+			$messages_data
+		);
+
+		$builder = new Prompt_Builder( AiClient::defaultRegistry(), array_values( $messages ) );
```

Comment:
Not sure why we wouldn't use this?

```suggestion
		$builder = AI_Client::prompt( array_values( $messages ) );
```

Reply:
Great catch, will fix it.

Reply:
Fixed!